stroom-core/src/main/java/stroom/core/query/SuggestionsService.java

```diff
@@ -1,10 +1,14 @@
 package stroom.core.query;
 
+import stroom.docref.DocRef;
 import stroom.query.shared.FetchSuggestionsRequest;
 
 import java.util.List;
+import java.util.function.Function;
 
 public interface SuggestionsService {
 
+    void registerHandler(final String dataSourceType, final SuggestionsQueryHandler handler);
```

Comment:
Our standard pattern for doing this sort of thing is to use a guice mapbinder, e.g. RestResourcesBinder, so each stroom module need only have a service that implements SuggestionQueryHander and a guice module that binds it to the datasourceType.

Reply:
Done